Index: hostap2/src/crypto/tls_openssl.c
===================================================================
--- hostap2.orig/src/crypto/tls_openssl.c	2014-06-11 10:47:01.277104988 +0300
+++ hostap2/src/crypto/tls_openssl.c	2014-06-11 10:47:01.273104988 +0300
@@ -765,6 +765,11 @@
 #endif /* OPENSSL_FIPS */
 #endif /* CONFIG_FIPS */
 		SSL_load_error_strings();
+		/* Only add potentially weak hashes and encryption algorithms
+		 * when FIPS mode is not enabled.
+		 */
+		if (!conf || !conf->fips_mode)
+			OpenSSL_add_all_algorithms();
 		SSL_library_init();
 #if (OPENSSL_VERSION_NUMBER >= 0x0090800fL) && !defined(OPENSSL_NO_SHA256)
 		EVP_add_digest(EVP_sha256());
