diff -Naur hostap.orig/wpa_supplicant/dbus/dbus_new_handlers.c hostap/wpa_supplicant/dbus/dbus_new_handlers.c
--- hostap.orig/wpa_supplicant/dbus/dbus_new_handlers.c	2014-05-12 13:27:01.351691716 +0300
+++ hostap/wpa_supplicant/dbus/dbus_new_handlers.c	2014-05-12 13:27:51.319689907 +0300
@@ -1533,16 +1533,6 @@
 
 	wpas_notify_network_removed(wpa_s, ssid);
 
-	if (wpa_config_remove_network(wpa_s->conf, id) < 0) {
-		wpa_printf(MSG_ERROR,
-			   "wpas_dbus_handler_remove_network[dbus]: "
-			   "error occurred when removing network %d", id);
-		reply = wpas_dbus_error_unknown_error(
-			message, "error removing the specified network on "
-			"this interface.");
-		goto out;
-	}
-
 	if (ssid == wpa_s->current_ssid)
 		wpa_supplicant_deauthenticate(wpa_s,
 					      WLAN_REASON_DEAUTH_LEAVING);
@@ -1553,6 +1543,16 @@
 		wpa_supplicant_req_scan(wpa_s, 0, 0);
 	}
 
+	if (wpa_config_remove_network(wpa_s->conf, id) < 0) {
+		wpa_printf(MSG_ERROR,
+			   "wpas_dbus_handler_remove_network[dbus]: "
+			   "error occurred when removing network %d", id);
+		reply = wpas_dbus_error_unknown_error(
+			message, "error removing the specified network on "
+			"this interface.");
+		goto out;
+	}
+
 
 out:
 	os_free(iface);
