Index: hostap2/wpa_supplicant/dbus/fi.w1.wpa_supplicant1.service.in
===================================================================
--- hostap2.orig/wpa_supplicant/dbus/fi.w1.wpa_supplicant1.service.in	2014-06-11 10:47:10.645104629 +0300
+++ hostap2/wpa_supplicant/dbus/fi.w1.wpa_supplicant1.service.in	2014-06-11 10:47:10.641104630 +0300
@@ -1,5 +1,5 @@
 [D-BUS Service]
 Name=fi.w1.wpa_supplicant1
-Exec=@BINDIR@/wpa_supplicant -u
+Exec=@BINDIR@/wpa_supplicant -B -u -f /var/log/wpa_supplicant.log -c /etc/wpa_supplicant/wpa_supplicant.conf -P /var/run/wpa_supplicant.pid
 User=root
 SystemdService=wpa_supplicant.service
Index: hostap2/wpa_supplicant/dbus/fi.epitest.hostap.WPASupplicant.service.in
===================================================================
--- hostap2.orig/wpa_supplicant/dbus/fi.epitest.hostap.WPASupplicant.service.in	2014-06-11 10:47:10.645104629 +0300
+++ hostap2/wpa_supplicant/dbus/fi.epitest.hostap.WPASupplicant.service.in	2014-06-11 10:47:10.641104630 +0300
@@ -1,5 +1,5 @@
 [D-BUS Service]
 Name=fi.epitest.hostap.WPASupplicant
-Exec=@BINDIR@/wpa_supplicant -u
+Exec=@BINDIR@/wpa_supplicant -B -u -f /var/log/wpa_supplicant.log -c /etc/wpa_supplicant/wpa_supplicant.conf -P /var/run/wpa_supplicant.pid
 User=root
 SystemdService=wpa_supplicant.service
